internal/gengapic/helpers.go

```diff
@@ -162,3 +163,48 @@ func isRequired(field *descriptor.FieldDescriptorProto) bool {
 
 	return false
 }
+
+// This takes in a path template from a routing annotation and converts it into a regex string.
+// The named capture is the named segment portion for the header itself.
+func convertPathTemplateToRegex(pattern string) string {
+	// If path template doesn't exist, then use a wildcard.
+	if len(pattern) < 1 {
+		return "(.*)"
+	}
+	// Replace name of header to named capture.
+	regexPattern := strings.ReplaceAll(pattern, "{", "(?P<")
+	regexPattern = strings.ReplaceAll(regexPattern, "}", ")")
+	// If not named, then entire segment is a wildcard.
+	if !strings.Contains(pattern, "=") || !strings.Contains(pattern, "/") {
+		regexPattern = strings.ReplaceAll(regexPattern, "*", "")
+		regexPattern = strings.ReplaceAll(regexPattern, "=", "")
+		regexPattern = strings.ReplaceAll(regexPattern, ")", ">.*)")
```

Comment:
Yup! Since this block of code only runs if the entire segment is a wildcard, line 181 will just replace the `)` to convert the entire thing to the named capture.